### PR TITLE
TRAPPIST-1 rescalings

### PR DESCRIPTION
**Pull request by eggrobin**
We may also need to do something about EVE clouds if @gregroxmun uses those.

### CODE REVIEW

File: tools/generate_kopernicus.cpp [hunk 1]
```diff
@@ -206,6 +225,25 @@ void GenerateKopernicusForSlippist1(
     kopernicus_cfg << "  }\n";
   }
   kopernicus_cfg << "}\n";
+  for (std::string const& name : solar_system.names()) {
+    bool const is_star =
+        !solar_system.keplerian_initial_state_message(name).has_parent();
+    if (is_star) {
```
Comment (pleroy) [inline tools/generate_kopernicus.cpp]: Preference for `if (!is_star) { ... }`. `continue` is a polite spelling of `goto`.

File: tools/generate_configuration.cpp [hunk 2]
```diff
@@ -79,7 +79,8 @@ void GenerateConfiguration(std::string const& game_epoch,
     serialization::GravityModel::Body const& body =
         solar_system.gravity_model_message(name);
     gravity_model_cfg << "  body {\n";
-    gravity_model_cfg << "    name                    = " << name << "\n";
+    gravity_model_cfg << "    name                    = "
+                      << (name == "Trappist-1" ? "Sun" : name) << "\n";
```
Comment (pleroy) [inline tools/generate_configuration.cpp]: Let's not hard-wire names.  Let's copy the code at lines 100-110 of `generate_kopernicus.cpp` to find the star (only if given by Keplerian elements) and let's use it here and below.

File: tools/generate_kopernicus.cpp [hunk 3]
```diff
@@ -148,6 +146,18 @@ void GenerateKopernicusForSlippist1(
...
+      for (char const* const curve :
+           {"ScaledIntensityCurve", "IntensityCurve", "IVAIntensityCurve"}) {
+        kopernicus_cfg << "        @" << curve << " {\n";
+        kopernicus_cfg << "          @key,*[0, ] *= 10\n";
+        kopernicus_cfg << "        }\n";
+      }
+      kopernicus_cfg << "      }\n";
+      kopernicus_cfg << "    }\n";
+    }
     if (!is_star) {
```
Comment (pleroy) [inline tools/generate_kopernicus.cpp]: `else`?

Comment (eggrobin) [reply tools/generate_kopernicus.cpp]: Done.

File: tools/generate_configuration.cpp [hunk 4]
```diff
@@ -163,7 +181,9 @@ void GenerateConfiguration(std::string const& game_epoch,
       auto const& body = barycentric_system.bodies[i];
       auto const& dof = barycentric_system.degrees_of_freedom[i];
       initial_state_cfg << "  body {\n";
-      initial_state_cfg << "    name = " << body->name() << "\n";
+      initial_state_cfg << "    name = "
+                        << (body->name() == "Trappist-1" ? "Sun" : body->name())
```
Comment (pleroy) [inline tools/generate_configuration.cpp]: This needs fixing too.

**Pull request merged**